spec/utils/crystal/memoization_spec.cr

```diff
@@ -0,0 +1,142 @@
+require "spec"
```

Comment:
These specs are copied with only minor modification from the `lucky` repo.

Reply:
One notable change is the `it "holds on to result of previous calls"` spec, which I renamed the description of (from `it "does not hold on to result of previous calls"` [link](https://github.com/luckyframework/lucky/blob/dd21f8bf9031ed444797534cab07fcbbb149e950/spec/lucky/memoize_spec.cr#L92C3-L92C52)) and substantively altered the content of, to reflect the behavior change that I implemented.